tests/test_nightReport.py

```diff
@@ -125,10 +125,23 @@ def test_rebuild(self):
     def test_getExposureMidpoint(self):
         """Test the exposure midpoint calculation
         """
-        midPoint = self.report.getExposureMidpoint(self.seqNums[0])
-        record = self.report.data[self.seqNums[0]]
-        self.assertGreater(midPoint, record['datetime_begin'].mjd)
-        self.assertLess(midPoint, record['datetime_end'].mjd)
+        # we would like a non-zero exptime exposure really
+        seqNumToUse = 0
+        for seqNum in self.report.data.keys():
+            expTime = self.report.data[seqNum]['exposure_time']
+            if expTime > 0:
+                seqNumToUse = seqNum
+                break
+
+        midPoint = self.report.getExposureMidpoint(seqNumToUse)
+        record = self.report.data[seqNumToUse]
+
+        if expTime == 0:
+            self.assertGreaterEqual(midPoint, record['datetime_begin'].to_datetime())
+            self.assertLessEqual(midPoint, record['datetime_end'].to_datetime())
+        else:
+            self.assertGreater(midPoint, record['datetime_begin'].to_datetime())
+            self.assertLess(midPoint, record['datetime_end'].to_datetime())
```

Comment:
If these are butler dimension records or something derived from them, it'd be good to move away from the `datetime_begin` and `datetime_end` nomenclature in favor of `timespan.begin` and `timespan.end` when you get a chance.  The `datetime_*` forms were grandfathered in and require some special-casing, and while we may never get around to deprecating the, I'd prefer to see `timespan.*` in new code.